src/pages/workspaces/workspace/analysis/Analyses.test.ts

```diff
@@ -0,0 +1,375 @@
+      await render(h(BaseAnalyses, defaultAnalysesProps))
+    })
+
+    // Assert
+    screen.getByText('Start')
+    // Banner text and images
+    screen.getByText('A place for all your analyses')
+    screen.getByAltText('Jupyter')
+    screen.getByAltText('RStudio Bioconductor')
+    screen.getByAltText('Galaxy')
+  })
+
+  it('loads properly with files for a google workspace', async () => {
+    // Arrange
+    const fileName1 = 'file1.ipynb'
+    const fileName2 = 'file2.ipynb'
+    const files = [getFileFromPath(`test/${fileName1}` as AbsolutePath), getFileFromPath(`test/${fileName2}` as AbsolutePath)]
+
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
+      ...defaultUseAnalysisStore,
+      loadedState: { state: files, status: 'Ready' }
+    }))
+
+    // Act
+    await act(async () => {
+      await render(h(BaseAnalyses, defaultAnalysesProps))
+    })
+
+    // Assert
+    screen.getByText('Start')
+    // Table headings
+    screen.getByText('Application')
+    screen.getByText('Name')
+    screen.getByText('Last Modified')
+    screen.getByPlaceholderText('Search analyses')
+    screen.getByText(fileName1)
+    screen.getByText(fileName2)
+  })
+
+  it('loads properly with files for an azure workspace', async () => {
+    // Arrange
+    const fileName1 = 'file1.ipynb'
+    const fileName2 = 'file2.ipynb'
+    const files = [getFileFromPath(`test/${fileName1}` as AbsolutePath), getFileFromPath(`test/${fileName2}` as AbsolutePath)]
+
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
+      ...defaultUseAnalysisStore,
+      loadedState: { state: files, status: 'Ready' }
+    }))
+
+    // Act
+    await act(async () => {
+      await render(h(BaseAnalyses, { ...defaultAnalysesProps, workspace: defaultAzureWorkspace }))
+    })
+
+    // Assert
+    screen.getByText('Start')
+    // Table headings
+    screen.getByText('Application')
+    screen.getByText('Name')
+    screen.getByText('Last Modified')
+    screen.getByPlaceholderText('Search analyses')
+    screen.getByText(fileName1)
+    screen.getByText(fileName2)
+  })
+
+  it('Sorts analysis table properly', async () => {
+    // Arrange
+    const fileName1 = 'file1.ipynb'
+    const fileName2 = 'file2.ipynb'
+    const file1Date = new Date().getTime()
+    const file2Date = file1Date + 10000
+    const files = [{ ...getFileFromPath(`test/${fileName1}` as AbsolutePath), lastModified: file1Date }, { ...getFileFromPath(`test/${fileName2}` as AbsolutePath), lastModified: file2Date }]
+
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
+      ...defaultUseAnalysisStore,
+      loadedState: { state: files, status: 'Ready' },
+    }))
+
+    // Act
+    await act(async () => {
+      await render(h(BaseAnalyses, defaultAnalysesProps))
+
+      // Assert base state
+      const analysisRowsBeforeSort: HTMLElement[] = await screen.findAllByRole('row')
+      // 2 files plus header row
+      expect(analysisRowsBeforeSort.length).toBe(3)
```

Comment:
nit:  these values can be stored for later assertion in a unified //Assert section at the end of the test... You already have a nice naming convention for things.

Reply:
same nit for a few tests after this.

Reply:
Hmm... not sure how I could get the initial htmlElements out of the `act` block scoping without having 2 separate `act`, since the render has to be in an act before I can grab the `beforeSort` portion from the DOM

Reply:
resolve down to the asserted value... here its number of rows === 3.... 

```
const numRowsBeforeSort = await screen.findAllByRole('row').length

...

// Assert
expect(numRowsBeforeSort).toBe(3)
```

Reply:
not a show-stopper... approving and letting you decide if you want to tweak the assert flow.